### PR TITLE
test: add jaegertracing support to our helm chart

### PR DESCRIPTION
**Pull request by tiagolobocastro**
fix: as it stands, core-agents should be a statefulset

---------

test: add jaegertracing support to our helm chart

It's deployed via the jaeger-operator which allows easy customization via
the Jaeger CRD.
Defaulted to using an in-memory database version, meant mostly for local tests.
It's disabled by default. To enable it use "--set jaeger: enabled: true"

### CODE REVIEW

File: chart/templates/jaeger-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Using Jaegertracing
```
Comment (paulyoong) [inline chart/templates/jaeger-operator/README.md:1]: Nice :+1:

File: deploy/csi-deployment.yaml [hunk 2]
```diff
@@ -22,6 +22,7 @@ spec:
       dnsPolicy: ClusterFirstWithHostNet
       imagePullSecrets:
         - name: regcred
+      initContainers:
```
Comment (paulyoong) [inline deploy/csi-deployment.yaml:25]: Is this automatically generated? We don't actually have any init containers here.

**Review by paulyoong — APPROVED** [review]
lgtm

File: deploy/csi-deployment.yaml (hunk 2 above)
Comment (tiagolobocastro) [reply deploy/csi-deployment.yaml:25]: Yes it is, it's empty because jaegertracing is disabled by default

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**